### DISCUSSION TITLE
Snowflake SQLAlchemy - Create table with Timestamp?

### DISCUSSION THREAD

**Discussion by BFASRachelS** [sqlalchemy/sqlalchemy#11484] [Usage Questions]
I am creating table definitions with SqlAlchemy in Python. I am able to successfully create a table with primary key which autoincrements and some other fields. However, I haven't been able to find any documentation on how to add a field with a timestamp which will be populated when a new record is added. For reference, this is how you would create this field directly in a Snowflake worksheet.

```
    CREATE OR REPLACE TABLE My_Table(
     TABLE_ID NUMBER NOT NULL PRIMARY KEY,
     ... Other fields
    TIME_ADDED TIMESTAMP DEFAULT CURRENT_TIMESTAMP
    );
```

This is my Python code...

 ```
   from sqlalchemy import Column, DateTime, Integer, Text, String
    from sqlalchemy.orm import declarative_base
    from sqlalchemy import create_engine
    
    Base = declarative_base()
    
    
    class My_Table(Base):
        __tablename__ = 'my_table'
    
        TABLE_ID = Column(Integer, primary_key=True, autoincrement=True)
        .. other columns
        .. Also need to create TIME_ADDED field
    
    
    engine = create_engine(
        'snowflake://{user}:{password}@{account_identifier}/{database}/{schema}?warehouse={warehouse}'.format(
            account_identifier=account,
            user=username,
            password=password,
            database=database,
            warehouse=warehouse,
            schema=schema
        )
    )
    
    Base.metadata.create_all(engine)
```

Has anyone done this and, if so, could you point me in the right direction?

**Comment by zzzeek** [comment]
we don't maintain the snowflake dialect on this end and I have a vague recollection that it has very awkward behaviors with timestamps, but overall you'd want to get help from the snowflake tracker at https://github.com/snowflakedb/snowflake-sqlalchemy

**Reply by BFASRachelS** [reply]
Thank you. I will try over there.
